qiskit/qpy/binary_io/schedules.py

```diff
@@ -71,6 +72,36 @@ def _loads_symbolic_expr(expr_bytes):
     return expr
 
 
+def _format_legacy_qiskit_pulse_v5(pulse_type, parameters):
+    # In the transition to Qiskit Terra > 0.22, the representation of library pulses was changed from
+    # complex "amp" to float "amp" and "angle". To reflect this, QPY version was bumped to 6. The
+    # existing library pulses in QPY<=5 are handled here separately to conform with the new
+    # representation. To avoid role assumption for "amp" for custom pulses, only the library pulses
+    # are handled this way.
+
+    # Note that parameters is mutated during the function call
+
+    # List of pulses in the library in QPY version 5 and below:
+    v5_library_pulses = ["Gaussian", "GaussianSquare", "Drag", "Constant"]
+
+    if pulse_type in v5_library_pulses:
+        # Once complex amp support will be deprecated we will need:
+        # parameters["angle"] = np.angle(parameters["amp"])
+        # parameters["amp"] = np.abs(parameters["amp"])
+        # _amp, _angle = sym.symbols("amp, angle")
+        # envelope = envelope.subs(_amp, _amp * sym.exp(sym.I * _angle))
```

Comment:
I think this line is needed to pass the qpy compatibility test. The reference pulse generated by dev version will have `sym.exp(sym.I * _angle)` and this check will fail.
https://github.com/Qiskit/qiskit-terra/blob/e775b4d19ad62558f30f39814b3d923ce03c8654/qiskit/pulse/library/symbolic_pulses.py#L554-L555

Reply:
@nkanazawa1989 
This is actually consistent with the rest of the code, and should pass the test. We only conditionally added the angle part:

        # To conform with some old tests, the angle part is inserted only when needed.
        if angle != 0:
            envelope_expr *= sym.exp(sym.I * _angle)

(line 667 of `symbolic_pulses.py` which I can't link for some reason).

So, it should pass the test. But in hindsight my reasoning seems very shaky. Maybe it's best to modify the old tests and always add the angle part. What do you think?

Reply:
Yes, I think that is much simpler to understand, but I don't have any strong suggestion here. Your explanation makes sense and change is up to you.

Reply:
I think you are right. I'll change it so all the pulses have the angle part in their envelopes. It will be simpler.